### ISSUE TITLE
Ús de tor/i2p per defecte

### ISSUE THREAD

**Issue by drym3r** [s3rrallonga/s3rrallonga.github.io#11]
Bones
He descarregat i examinat el codi per que em sembla molt interessant la forma de fer servir un programa d'aquest tipus per organitzar a la gent. 
Lo primer es dir que no tinc gaire idea ni de java ni de kotlin. He vist que está basat en retroshare, una red p2p (o f2f com diuen en un altre issue). M'ha preocupat una mica no veure cap referencia a i2p o tor al codi, lo cual em fa pensar que no s'està fent servir.  Entenc que es confia en que la gent compartirà acces nomès a cercles de confiança, pero tenint un ús massiu em sembla ingenu creure que això passarà. 
Les conseqüencies d'aquesta expoisiciò sería que qualsevol membre de la red pot tenir una llista de totes les IP que participen a la red de Tsunami Democratic. 
Fa servir ja alguna red d'anonimitzaciò l'aplicaciò? Si no, n'hi donareu suport en algún moment? 
En la descripciò diu que l'app es anonima, però em sembla que no ho es de veritat.
Saluts i felicitats pel treball

**Comment by retroixer** [comment]
A #9 defenso per què no caldria Tor ni I2P si estan aprofitant a fons libretroshare. En resum, crec que l'adreça IP de cada mòbil només s'està compartint amb un node intermediari associat a cada QR. Aquests nodes intermediaris serien reemplaçables i constituirien el teixit de darknet entre activistes seguidors i activistes impulsors.

Estaria bé que l'equip @s3rrallonga justifiqués una mica més les reclamacions de "anònima" i el funcionament perquè puguem decidir amb criteri si confiar en la plataforma, i quins riscos comporta realment.

**Comment by sergivb01** [comment]
En aquesta xerrada de FOSDEM s'explica el funcionament de RetroShare https://www.youtube.com/watch?v=FR9SSdGN0K8

**Comment by drym3r** [comment]
No estic d'acord en que es compleixi @retroixer, tot i suposant que es faci. El tema es que una red basada en la confiança necessita que la gent sapiga el que fa. Si no ho sap, com es el cas donat que s'abstrau a la gent de com funciona l'aplicació, no es pot confiar en que la gent prengui decisions. 
I encara que es fes, hi ha un altre vector d'atac molt simple del que no protegeix, i es el de la incautacio de móvils. Si algú es detingut, qualsevol conegut de la red passaria a ser identificat automàticament.

**Comment by retroixer** [comment]
> En aquesta xerrada de FOSDEM s'explica el funcionament de RetroShare https://www.youtube.com/watch?v=FR9SSdGN0K8

Sí! És una mica avorrida, però explica el paper amb paraules més humanes. Al FOSDEM hi tenen més informació, i a més, el vídeos no els allotgen a Youtube, que s'agraeix molt.
* [FOSDEM 2018 - The Generic Data Distribution System of the Retroshare Network](https://archive.fosdem.org/2018/schedule/event/retroshare/)
* [FOSDEM 2019 - Retroshare JSON API](https://archive.fosdem.org/2019/schedule/event/retroshare_json/)

**Comment by retroixer** [comment]
> No estic d'acord en que es compleixi @retroixer, tot i suposant que es faci. El tema es que una red basada en la confiança necessita que la gent sapiga el que fa.

Estic d'acord amb la idea de fons, però crec que les accions que s'exposen a nivell gràfic són tan poques, que l'únic forat de confiança que veig possible és la mala gestió dels QR.

> Si no ho sap, com es el cas donat que s'abstrau a la gent de com funciona l'aplicació, no es pot confiar en que la gent prengui decisions.

Si no sabem (ni "la gent" ni nosaltres) què hi ha al QR ben bé, què comporta que una entitat atacant aconsegueixi un QR... Si només s'haurien d'enviar a persones amb qui compartiries roba interior, o a qualsevol que et mereixi una mínima confiança ja és suficient. Què està en risc?

> I encara que es fes, hi ha un altre vector d'atac molt simple del que no protegeix, i es el de la incautacio de móvils. Si algú es detingut, qualsevol conegut de la red passaria a ser identificat automàticament.

Això no ho podem sabem encara, però espero que t'equivoquis. A què et refereixes amb "qualsevol conegut de la red"? Crec que és la clau.

A mi em sembla (i espero) que l'únic node conegut per cada mòbil serà un node "servidor" corrent en un VPS a un país amb jurisdiccions especials.

Però per això crec que @s3rrallonga haurien d'anar explicant com funciona el sistema i publicar el codi i el disseny perquè tothom pugui prendre decisions informades. Saber si cal una VPN o Tor o no, els riscos i garanties... No és una web d'un dia com el 1-O, si hem de creure el que diuen i això va per llarg, cal un procés d'adaptació, informació, confiança, etc.

**Comment by joanotpetitot** [comment]
> Si no sabem (ni "la gent" ni nosaltres) què hi ha al QR ben bé, què comporta que una entitat atacant aconsegueixi un QR... Si només s'haurien d'enviar a persones amb qui compartiries roba interior, o a qualsevol que et mereixi una mínima confiança ja és suficient. Què està en risc? 

Simple especulació, pero pel que crec que se sobre com funciona introduir un QR infiltrat no seria precisament facil. Pensa com si algú vulgués introduir una clau RSA a una oficina per entrar als servidors que hi ha al CPD. Si no tens les dues parts de la clau és practicament impossible aconseguir que una clau externa serveixi per autenticarse i entrar als servidors.

Per altra banda, les IPs pel que fa a telefonia movil estan totes cobertes per un NAT enorme que la operadora utilitza per no gastar IPs publiques sense necessitat. Centenars de dispositius tenen, de cara enfora, la mateixa IP, sobretot si estan en una zona geografica limitada i propera. Es possible arribar a distingir qui va fer què amb tal dispositiu, IP, hora, etc, pero en un cas com aquest que inclou centenars de persones que simplement s'han descarregat una aplicació de missatgeria amb tota la bona voluntat de manifestarse de manera pacifica es una perdua de temps enorme amb poca o cap validesa juridica.

Crec, vaja. Tot es pura especulació, ja em corregeireu si veieu que cal.

**Comment by retroixer** [comment]
> > Si no sabem (ni "la gent" ni nosaltres) què hi ha al QR ben bé, què comporta que una entitat atacant aconsegueixi un QR... Si només s'haurien d'enviar a persones amb qui compartiries roba interior, o a qualsevol que et mereixi una mínima confiança ja és suficient. Què està en risc?
> 
> Simple especulació, pero pel que crec que se sobre com funciona introduir un QR infiltrat no seria precisament facil. Pensa com si algú vulgués introduir una clau RSA a una oficina per entrar als servidors que hi ha al CPD. Si no tens les dues parts de la clau és practicament impossible aconseguir que una clau externa serveixi per autenticarse i entrar als servidors.

Crec que no ens hem entès. No em referia a que una entitat atacant aconseguís generar un QR propi, que és el que em sembla que dius. Em referia a quines conseqüències pot tenir que aquesta entitat aconsegueixi un QR bo. Quines dades hi ha codificades? Què exposa? Les persones que hagin afegit aquest QR, què? No sabem quin perill hi ha.

> 
> Per altra banda, les IPs pel que fa a telefonia movil estan totes cobertes per un NAT enorme que la operadora utilitza per no gastar IPs publiques sense necessitat. Centenars de dispositius tenen, de cara enfora, la mateixa IP, sobretot si estan en una zona geografica limitada i propera.

Sí, però també cal tenir en compte el "cara cap endins". Aquest "dins" són companyies telefòniques sota la legislació espanyola, i per tant, col·laboradores de la vigilància estatal.
